config/v1/types_authentication.go

```diff
@@ -222,11 +223,21 @@ type OIDCProvider struct {
 	//
 	// +listType=atomic
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// UserValidationRules defines the set of rules used to validate claims in a user’s token.
+	// These rules determine whether a token subject is considered valid based on its claims.
+	// Each rule is evaluated independently.
+	// See the TokenUserValidationRule type for more information on rule structure.
+	// +optional
+
+	//+listType=atomic
+	UserValidationRules []TokenUserValidationRule `json:"userValidationRules,omitempty"`
```

Comment:
We also need to gate all the new fields with the new feature gate you are adding. Otherwise, it will show up by default in all the feature-gates. Because the `ExternalOIDC` feature-gate is already GA in HyperShift, that would make this field immediately GA - which we don't want.
```suggestion
	// +openshift:enable:FeatureGate=ExternalOIDCWithNewAuthConfigFields
	UserValidationRules []TokenUserValidationRule `json:"userValidationRules,omitempty"`
```